File: internal/domain/ptszi.go
package domain

type PTSZIThreat struct {
	ID          int64    `json:"id"`
	Code        string   `json:"code"`
	Name        string   `json:"name"`
	Description *string  `json:"description,omitempty"`
	QThreat     float64  `json:"q_threat"`
	QSeverity   float64  `json:"q_severity"`
	Contours    []string `json:"contours,omitempty"`
}

type PTSZIVulnerableLink struct {
	ID          int16   `json:"id"`
	Code        string  `json:"code"`
	Name        string  `json:"name"`
	Description *string `json:"description,omitempty"`
}

type PTSZIControl struct {
	ID          int16   `json:"id"`
	Code        string  `json:"code"`
	Name        string  `json:"name"`
	Description *string `json:"description,omitempty"`
}

type PTSZIAssetVulnerableLink struct {
	VulnerableLink PTSZIVulnerableLink `json:"vulnerable_link"`
	Status         string              `json:"status"`
	Comment        *string             `json:"comment,omitempty"`
}

type PTSZIAssetControl struct {
	Control       PTSZIControl `json:"control"`
	Effectiveness float64      `json:"effectiveness"`
	Comment       *string      `json:"comment,omitempty"`
}

type PTSZIUBIThreat struct {
	ID                    int64    `json:"id"`
	UBICode               string   `json:"ubi_code"`
	UBINumber             int      `json:"ubi_number"`
	Name                  string   `json:"name"`
	Description           *string  `json:"description,omitempty"`
	SourceRaw             *string  `json:"source_raw,omitempty"`
	ImpactObject          *string  `json:"impact_object,omitempty"`
	ImpactConfidentiality bool     `json:"impact_confidentiality"`
	ImpactIntegrity       bool     `json:"impact_integrity"`
	ImpactAvailability    bool     `json:"impact_availability"`
	MaxPotential          string   `json:"max_potential"`
	QThreat               float64  `json:"q_threat"`
	QSeverity             float64  `json:"q_severity"`
	MappedSources         []string `json:"mapped_sources,omitempty"`
}

type PTSZIAttackPath struct {
	Asset              AssetRef              `json:"asset"`
	AssetContour       string                `json:"asset_contour"`
	Threat             PTSZIThreat           `json:"threat"`
	Sources            []ThreatSource        `json:"sources"`
	VulnerableLinks    []PTSZIPathVL         `json:"vulnerable_links"`
	DestructiveActions []DestructiveAction   `json:"destructive_actions"`
	UBI                []PTSZIUBIThreat      `json:"ubi"`
	QThreat            float64               `json:"q_threat"`
	QSeverity          float64               `json:"q_severity"`
	QReaction          float64               `json:"q_reaction"`
	Z                  float64               `json:"z"`
	W                  float64               `json:"w"`
	Level              string                `json:"level"`
	Applicable         bool                  `json:"applicable"`
	MissingControls    []PTSZIControl        `json:"missing_controls"`
	Recommendations    []PTSZIRecommendation `json:"recommendations,omitempty"`
}

type PTSZIPathVL struct {
	VulnerableLink PTSZIVulnerableLink    `json:"vulnerable_link"`
	Status         string                 `json:"status"`
	Comment        *string                `json:"comment,omitempty"`
	Coverage       float64                `json:"coverage"`
	Uncovered      bool                   `json:"uncovered"`
	Controls       []PTSZIControlCoverage `json:"controls"`
}

type PTSZIControlCoverage struct {
	Control           PTSZIControl `json:"control"`
	Coverage          float64      `json:"coverage"`
	Implemented       bool         `json:"implemented"`
	Effectiveness     float64      `json:"effectiveness"`
	ResultingCoverage float64      `json:"resulting_coverage"`
}

type PTSZIRecommendation struct {
	ControlID   int16  `json:"control_id"`
	ControlCode string `json:"control_code"`
	Category    string `json:"category"`
	Title       string `json:"title"`
	Description string `json:"description"`
	Priority    string `json:"priority"`
}

type PTSZIAssetProfile struct {
	Asset             Asset                      `json:"asset"`
	SecurityContour   string                     `json:"security_contour"`
	VulnerableLinks   []PTSZIAssetVulnerableLink `json:"vulnerable_links"`
	Controls          []PTSZIAssetControl        `json:"controls"`
	ApplicableThreats []PTSZIAttackPath          `json:"applicable_threats"`
}

File: frontend/src/types/ptszi.ts
import { Asset } from "../types";

export interface ThreatSource {
  id: number;
  code: string;
  name: string;
  description?: string;
}

export interface PtsziThreat {
  id: number;
  code: string;
  name: string;
  description?: string;
  q_threat: number;
  q_severity: number;
  contours?: string[];
}

export interface PtsziVulnerableLink {
  id: number;
  code: string;
  name: string;
  description?: string;
}

export interface PtsziControl {
  id: number;
  code: string;
  name: string;
  description?: string;
}

export interface DestructiveAction {
  id: number;
  code: string;
  name: string;
  affects_confidentiality: boolean;
  affects_integrity: boolean;
  affects_availability: boolean;
}

export interface PtsziUBIThreat {
  id: number;
  ubi_code: string;
  ubi_number: number;
  name: string;
  description?: string;
  source_raw?: string;
  impact_object?: string;
  impact_confidentiality: boolean;
  impact_integrity: boolean;
  impact_availability: boolean;
  max_potential: string;
  q_threat: number;
  q_severity: number;
  mapped_sources?: string[];
}

export interface PtsziControlCoverage {
  control: PtsziControl;
  coverage: number;
  implemented: boolean;
  effectiveness: number;
  resulting_coverage: number;
}

export interface PtsziPathVL {
  vulnerable_link: PtsziVulnerableLink;
  status: string;
  comment?: string;
  coverage: number;
  uncovered: boolean;
  controls: PtsziControlCoverage[];
}

export interface PtsziRecommendation {
  control_id: number;
  control_code: string;
  category: string;
  title: string;
  description: string;
  priority: string;
}

export interface PtsziAttackPath {
  asset: { id: number; name: string };
  asset_contour: string;
  threat: PtsziThreat;
  sources: ThreatSource[];
  vulnerable_links: PtsziPathVL[];
  destructive_actions: DestructiveAction[];
  ubi: PtsziUBIThreat[];
  q_threat: number;
  q_severity: number;
  q_reaction: number;
  z: number;
  w: number;
  level: "critical" | "high" | "medium" | "low";
  applicable: boolean;
  missing_controls: PtsziControl[];
  recommendations?: PtsziRecommendation[];
}

export interface PtsziAssetProfile {
  asset: Asset;
  security_contour: string;
  vulnerable_links: Array<{
    vulnerable_link: PtsziVulnerableLink;
    status: string;
    comment?: string;
  }>;
  controls: Array<{
    control: PtsziControl;
    effectiveness: number;
    comment?: string;
  }>;
  applicable_threats: PtsziAttackPath[];
}
